src/Runner.Listener/MessageListener.cs

```diff
@@ -206,6 +228,11 @@ public async Task<TaskAgentMessage> GetNextMessageAsync(CancellationToken token)
                         continuousError = 0;
                     }
                 }
+                catch (OperationCanceledException) when (_getMessagesTokenSource.Token.IsCancellationRequested)
```

Comment:
`_getMessagesTokenSource.Token.IsCancellationRequested` will always be true when `token` is cancelled

I think you need to do
`_getMessagesTokenSource.Token.IsCancellationRequested && !token.IsCancellationRequested`

Reply:
updated pr